### DISCUSSION TITLE
ASA Stats Deep Liquidity issue

### DISCUSSION THREAD

**Discussion by wxivd** [asastats/channel#62] [General]
### Main Highlights
* Modest liquidity in Tinyman Algo/ASA Stats LP pool hindering large investments & New investors. Referring to recent Discord [discussion](https://discord.com/channels/906917846754418770/910524302531649566/971303919877763133).
* Single main Tinyman Algo/ASA Stats LP pool, hindering options to utilise other Algorand AMMs. Some context has been [discussed](https://discord.com/channels/906917846754418770/914540797054025788/952580397009625118) before.

Increasing ASA Stats liquidity can be mainly achieved by:
* ASA Stats project fund
* Investors

There are many options that I want to discuss/propose, these aim to:
* Secure deep liquidity, increase trading volume / attract new investments
* Increase viability of utilising other LP pools on Algorand
* Raise capital to serve all prior points and accelerate project development

### A: ASA Stats LP rewards incentive program: Proposed update
**Background**

- [Rough Draft for LP Provider Incentives](https://www.reddit.com/r/asastats/comments/s9m3p2/rough_draft_for_lp_provider_incentives/?utm_source=share&utm_medium=web2x&context=3)
- [ASA Stats is officially rolling out proposed LP rewards plan](https://www.reddit.com/r/asastats/comments/slm2b4/asa_stats_is_officially_rolling_out_proposed_lp/)
- Current [LP rewards incentive program](https://www.reddit.com/r/asastats/comments/svm1qa/refinement_of_weekly_asastats_lp_rewards/)

The ASA Stats LP rewards wallet activity can be reviewed here: VZB3AHTAIBWEH763WMZNIZGZNOM3XYJTQ3AHUJF2YDMM525RRFIHQT3XAQ

> To qualify for rewards, a community member must hold Tinyman ASASTATS/ALGO LP tokens for a period of 7 days from the initial snapshot, until the next snapshot occurring on Sunday 00:00 UTC.

> Flat 3 ASASTATs reward per LP token, with an entry floor of 800 LP tokens and a ceiling of 9000. If the overall weekly limit is not encountered, this would translate into a weekly distribution between 2400-27000 ASASTATs for all qualifying wallets.

> To adhere to the original plan of providing this incentive for a total of 104 weeks, the limit of the weekly payout will continue to be 480,769.23. Once this limit is encountered, the rewards ratios will reduce in a fair and relational manner.

**Proposed update: Incentivise ASA Stats investors/token holders to provide more liquidity**
1. A: Reduce the length of the program to 52 Weeks (1 Year) to provide more LP rewards. Increase the LP tokens ceiling accordingly (from 9000 to optimised amount). 

- The duration of the program may have been aligned to the duration of team-run ASA Stats until the transition to DAO phase. However, this should not be the only driving factor. The first year of the project builds the foundation for success, hence the proposed update. In this upcoming year our MVP would be live brining token utility/generating profit.
- The proposed duration of 52 weeks is long enough to account for Impermanent loss, as it provides higher return on investment.

2. A: Payouts of LP rewards is currently done by the team, we shall consider Algostake for increased visibility. Tinyman Algo/ASA Stats LP rewards on Algostake.

### B: ASA Stats Staking program
Yieldly announcement is anticipated anytime now
* YLDY - ASASTATS pool.
* ASASTATS - YLDY LP pool.
* ASASTATS - ASASTATS distribution pool.

These bring numerous advantages mainly,
* Increased visibility of ASA Stats project.
* Passive income for ASA Stats token holders: ASA Stats-ASA Stats distribution pool. Addressing high demand.
* Incentives for new investors to invest in ASA Stats bringing price stability and appreciation.

However this does not directly address the modest current liquidity in Tinyman Algo/ASA Stats LP pool, which leads us to C.

### C: ASA Stats NFT Collection
Creating our first NFT collection. This is a dual solution aims to increase liquidity by ASA Stats project fund and keen investors. 

1. C: **NFT advantages for ASA Stats project**
- Raise funds from NFT sales to provide deep liquidity to our LP pools (current/future). Expediting project development.
- Giveaway for active community members incentivising community contribution. 
- Marketing, Increase awareness of ASA Stats.

2. C: **NFT advantages/Utility for investors/community members**
- Staking Rewards for NFT holders: APR to holders paid in ASA Stats Tokens. Giving rewards to NFT holders is similar to liquidity provider incentive but we get to decide fund allocation in LP pools. NFT staking rewards runs for the duration of LP program.
- Free NFT drop to LP providers that satisfy additional criteria: X amount of LP tokens above ceiling limit + LP locked for specific duration
- Free advanced ASA Stats features to choose from once these are rolled out (1 free advanced feature per wallet, non-commercial use).

3. C: **NFT Creation process**
- Assign Budget
NFT creation Cost, the blockchain require to have 0,1A per NFT blocked on the creator's account + 0,001 for creating the asset. 
Artist commission: depends on reputation, ~month salary where he's come from, and
- Number of NFTs
- NFT Theme (art technique, traits)

**Initial thoughts**
- Estimated total cost: 1500-3000 Algo. We can negotiate a deal that include % of the sales for artist, a given number of NFTs (thus optimising cost).
- Total number 350 NFTs
- Minting price 150-200A
- NFT Theme, Generative art,  Stats for Statistics: have the following traits:
Category: Descriptive Statistics
Types: measures of frequency, central tendency, dispersion or variation, position.
Visual art (graphs, charts, pies, histograms, scatter plots..etc)
Description: basic statistics (educational value)

**Comment by ipaleka** [comment]
How eventual Algostake program can affect everything? https://github.com/asastats/channel/discussions/60

**Reply by wxivd** [reply]
Algostake has the following staking pools: 

- PRO
- LP
- NFT
- Asset

### LP
I think we start with **LP** pool (Tinyman Algo-ASA Stats). Two goals: 1: Highlight our LP reward program, 2: More visibility to ASA Stats project. 

> How eventual Algostake program can affect everything?

It is logical to expect a positive change in ASA Stats liquidity volume (Tinyman Algo-ASA Stats pool). If large enough, we may not need to change the duration of the ASA Stats LP rewards incentive program (Point 1.A). 

### ASA Stats Staking
**Asset staking** on Algostake may be considered in the future (ASA Stats-ASA Stats), ideally after the conclusion of staking program with Yieldly. I see no added advantage in running two ASA Stats staking pools simultaneously.

**Reply by ipaleka** [reply]
> ### LP
> 
> I think we start with **LP** pool (Tinyman Algo-ASA Stats Token). Two goals: 1: Highlight our LP reward program, 2: More visibility to ASA Stats project.

After we get [more information from them](https://github.com/asastats/channel/discussions/60#discussioncomment-2721615), we can discuss everything, eventually the amount of staking funds to allocate for the pool.

**Reply by ipaleka** [reply]
Btw, I should say that my main concern is, afaik, a lack of smart contracts on their side.

We can afford the same in our rewards incentive program, but I'm not really sure about that in the case of third parties.

**Reply by wxivd** [reply]
> Btw, I should say that my main concern is, afaik, a lack of smart contracts on their side.
> 
> We can afford the same in our rewards incentive program, but I'm not really sure about that in the case of third parties.

We can set up reward pool at Algostake in a way to cover LP rewards for short duration of time (1 million ASA Stats rewards for 2 weeks or so) and top it up as time goes by.

**Comment by AlgoRhythMatic** [comment]
> Increase the LP tokens ceiling accordingly (from 9000 to optimised amount).

+1 for this approach.  Is there a new ceiling or tiering structure you have in mind?

**Reply by wxivd** [reply]
I researched some ideas from other projects which I share with you here and discuss ASA Stats proposal. 

### DopePana 
* Total supply: 1B, Similar to ASA Stats
* Distribution method: Algostake 
* [LP Reward program](https://dopepanda.medium.com/moving-away-from-lp-competition-stake-your-algo-dpanda-lp-tokens-with-algostake-801557fb7caa): Tinyman Dpanda-Algo LP pool
    * 45M Dpanda yet to be distributed over next 7 months 
    * No ceiling on LP tokens
    * Flat rate per LP token decrease, currently 125 Dpanda per LP token (APR 96.57%). This is decreased to 25 Dpanda per LP token by the conclusion of the program.

**Comments 1**
- No ceiling criteria primes LP providers to invest more.
- Decreasing pay outs reward early investors, prioritising now.
- Large rewards may create selling pressure. Creating profile tiers with more privilege for higher levels (e.g. Pine, wallet hold minimum of 500K Dpanda) could potentially account for that.

### Algoneer 
* Total supply: 10M
* Distribution method: in house 
* [LP reward program](https://www.reddit.com/r/algoneer/comments/umn0fg/liquidity_pool_rewards_program_is_live/?utm_source=share&utm_medium=ios_app&utm_name=iossmf): Tinyman AGNR-Algo LP pool 
    * 10% of total supply
    * Rewards occur daily, distribution happen at random interval (1-100 days)
    * Amount of distribution depends on circulation supply LP tokens
    * User Reducing LP tokens > reduce rewards accordingly or reward lost for entire duration since previous distribution

**Comments 2** 
- Random periodic distribution encourages LP token holders to leave LP tokens.
- Backed by science, Skinner’s Operant Conditioning, Positive Reinforcement Theory. Through operant conditioning, an individual makes an association between a particular behaviour and a consequence. Schedules of Reinforcement (i.e. Rewards) is key:
- A: Continuous reinforcement: This method is impractical to use, Response rate is slow and the reinforced behavior is prone to extinction. **However, In cryptospace, investors act based on relativity: We tend to make decisions based on comparisons i.e which tokens gives more rewards.**
- B: Interval/ration schedule reinforcement yields slow but better response. **This is what Algonner is doing. Aiming for investors with long term mindset.**

### ASA Stats

- [TVL](https://app.tinyman.org/#/analytics) Tinyman ASA Stats-Algo Pool:  34.31M ASA Stats Token, 12.01K ALGO
- [Rewards reserve](https://www.asastats.com/VZB3AHTAIBWEH763WMZNIZGZNOM3XYJTQ3AHUJF2YDMM525RRFIHQT3XAQ): 43,287,380M ASA Stats token
- [LP providers](https://algoexplorer.io/asset/552659276): 53 LP provider satisfy the eligibility criteria (minimum 800 LP token). 

> Flat 3 ASASTATs reward per LP token, with an entry floor of 800 LP tokens and a ceiling of 9000. If the overall weekly limit is not encountered, this would translate into a weekly distribution between 2400-27000 ASASTATs for all qualifying wallets.

Maximum weekly distribution per wallet has not reached the 27000 max per wallet (ceiling of 9000 LP tokens). It is around 15K per wallet per week (therefore, actual rewards around 2 ASA Stats per LP tokens).

> +1 for this approach. Is there a new ceiling or tiering structure you have in mind?

My Idea has two parts

**Part one**
Aims to distribute the maximum of 30M ASA Stats token by the end on May 2023 (12 Month program) creating incentives for **more LP providers**, **now**.

1- Remove LP ceiling
2- Increase LP rewards from current actual rate (2 ASA Stats token per LP token) to X% APR per LP token (we need to calculate this, at 45%, 25%, 20%, 15, 12%, 6%) for optimisation and decision making.
3- Adopt distribution levels, LP Rewards decreased once the reward reserve for each level is distributed. 
- This gives us control of the duration of LP rewards program 
- Allows initial high APR to attract new investors, reward early LP providers. 

**Part two**
Aims to distribute remaining reserve that was set aside in part one, 13M ASA Stats. Attracting/rewarding investors with long-term mindset.

* In-house distribution plan
* Provide extra LP rewards based on **A:** total days LP provided and **B:** on average daily pool token. Idea stems/modified from [Nekoin](https://www.nekoin.ai/liquidity)
* Rewards depend on **Rank** and **duration**: Rank top 50 LP providers by (number of days, number of LP tokens with minimum 5K LP tokens), define periods (3 monthly), set rewards based on overall program duration
* Random LP Reward distribution creating positive reinforcement
* Duration of program open to discussion as everything mentioned

**Reply by ipaleka** [reply]
Thank you mate for, as always, a detailed and valuable analysis.

I should say that the amount you refer to as "Rewards reserve" is completely arbitrary. The airpool's total allocation is 80,000,000 ASASTATS and we defined that 25% of that amount (20,000,000 ASASTATS) goes for our incentive program:

https://discord.com/channels/906917846754418770/914540797054025788/939192274724745227
> For the Gaming initiative we'll start with 20,000,000 ASASTATS from the Airdrop Pool, and after that the Airdrop Pool, the Community Rewards Pool, the Administration Pool and eventually the Developer Pool can add up.

**Comment by ipaleka** [comment]
From now on until a resolution, this represents the most important discussion in ASA Stats. I propose that we link it to our community channels emphasizing that fact.

@SCN9A I suggest that we check with GamingToops and @dragmz for a possibility and/or justification for implementing your suggestions into our incentive program. If we can't manage to find a resolution that way, I guess that will automatically open a new discussion about the need for abandoning our incentive program and switching to Algostake instead.

**Reply by ipaleka** [reply]
@AlgoRhythMatic, @Babbexx22,  @bear1bear2bear3, @Damolicious, @moloch10, @motuwagon

**Reply by AlgoRhythMatic** [reply]
> From now on until a resolution, this represents the most important discussion in ASA Stats. I propose that we link it to our community channels emphasizing that fact.

I agree that we need to get more of the community involved in this discussion ASAP.  I will think about a message to present in Reddit to try and get people to respond w/ dialog.

In anticipating some reactions in liquidity discussions, I am thinking about GamingToups comments from Discord back in April: https://discord.com/channels/906917846754418770/914540797054025788/957350038382735471 

> Until utility is live additional rewards are likely to cause additional price dumps.  On one hand it's nice because those providing LP are down a good 50% or so (myself included) so some rewarding helps the feel bad but there is no guarantee those who receive it would hold.

Along with focus on considering improvements to liquidity, I think we should try to continue also with MVP considerations + efforts to increase our amount of partnership opportunities in a very visible way.

**Reply by dragmz** [reply]
What would be the point of developing own incentive program further if AlgoStake is sufficient?

**Reply by ipaleka** [reply]
> Along with focus on considering improvements to liquidity, I think we should try to continue also with MVP considerations + efforts to increase our amount of partnership opportunities in a very visible way.

Absolutely! This discussion is the most important, but it shouldn't take long, and afterward our priority moves to API MVP planning.

And after we end up with a plan and draft design for the MVP, respectively after the development starts, then the community focus should shift completely to marketing campaign planning/designing.

**Reply by ipaleka** [reply]
> What would be the point of developing own incentive program further if AlgoStake is sufficient?

Dunno, really, that's why we should discuss that.

Is Algostake sufficient for everything @SCN9A suggested? If it is then I suppose our first move should be to stake the rest of our Tinyman's ASASTATS-YLDY LP tokens in the related Yieldly's staking program ([comment1](https://github.com/asastats/channel/discussions/102#discussioncomment-2815742), [comment2](https://github.com/asastats/channel/discussions/102#discussioncomment-2819626)) in order to prevent a need for selling ASASTATS by admpool and/or further borrowing from the devpool.

**Reply by ipaleka** [reply]
> > What would be the point of developing own incentive program further if AlgoStake is sufficient?
> 
> Dunno, really, that's why we should discuss that.
> 
> Is Algostake sufficient for everything @SCN9A suggested? If it is then I suppose our first move should be to stake the rest of our Tinyman's ASASTATS-YLDY LP tokens in the related Yieldly's staking program ([comment1](https://github.com/asastats/channel/discussions/102#discussioncomment-2815742), [comment2](https://github.com/asastats/channel/discussions/102#discussioncomment-2819626)) in order to prevent a need for selling ASASTATS by admpool and/or further borrowing from the devpool.

Based on a bit of urgency and on [this comment](https://github.com/asastats/channel/discussions/102#discussioncomment-2836332), the [allocations were made](https://github.com/asastats/channel/discussions/102#discussioncomment-2836603).

**Reply by AlgoRhythMatic** [reply]
> What would be the point of developing own incentive program further if AlgoStake is sufficient?

So a bit late on this one but one huge difference (unless I am missing something):
- Algostake rewards holding ASASTATS
- Our program rewards ALGO/ASASTATS LP tokens (e.g. providing liquidity).

Massive difference there.  If we get rid of our current program w/o implementing a specific Algostake replacement program for ALGO/ASASTATS LP tokens, our main liquidity dries up at the speed of light - why should I provide LP when I get better incentives just holding?

**Reply by wxivd** [reply]
> Massive difference there. If we get rid of our current program w/o implementing a specific Algostake replacement program for ALGO/ASASTATS LP tokens, our main liquidity dries up at the speed of light - why should I provide LP when I get better incentives just holding?

I think @dragmz is addressing ASASTATS/Algo LP rewards on Algostake VS further development of own incentive program (distribution).

**Comment by ipaleka** [comment]
> C: ASA Stats NFT Collection

What do you think of creating a separate discussion for this and linking it in the initial post body @SCN9A ? 

Meanwhile, I'm going to c/p two comment from Discord:

- - - - -
by @motuwagon in [Discord](https://discord.com/channels/906917846754418770/973245225680109588/979733486686572544);
> We were talking about a Asa Stats nft a while back.  I am not a artist but anyone like the idea of Alpaca  nft as a play on the team leads name.

- - - - -
by Mayor | CM YouTube 
https://discord.com/channels/906917846754418770/973245225680109588/979767920240496690
> I'm interested in working with @ipaleka on an ASA series

- - - - -

**Reply by Motuwagon** [reply]
Quick google search there is a eth nft Alpaca project and a financial crypto company that uses name and mascot Alpaca and advertising their api for stock and crypto.  so maybe it would be a poor choice.  But did not find a Algo one yet.  Image search they are cute at any age.  With unique hair cut  can make for fun personality.
![Screenshot_20220527-152736](https://user-images.githubusercontent.com/103940952/170777604-17dbd766-efa2-4cd8-801d-fb9d8bec1f39.png)

**Reply by wxivd** [reply]
Discussion about ASA Stats NFT Collection has moved to [#111](https://github.com/asastats/channel/discussions/111)

**Comment by ipaleka** [comment]
Here's another provider, GamingToops pointed at it

https://thealgofaucet.com/staking/

If someone can investigate...

**Comment by RanTask** [comment]
Hey guys, pls lets migrate liquidity to tinyman v2 asap.

**Reply by ipaleka** [reply]
> Hey guys, pls lets migrate liquidity to tinyman v2 asap.

https://github.com/asastats/channel/discussions/328#discussioncomment-4603114
